modules/consensus/accept.go

```diff
@@ -207,78 +195,110 @@ func (cs *ConsensusSet) addBlockToTree(b types.Block) (ce changeEntry, err error
 // This method is typically only be used when there would otherwise be multiple
 // consecutive calls to AcceptBlock with each successive call accepting the
 // child block of the previous call.
-func (cs *ConsensusSet) managedAcceptBlock(b types.Block) error {
-	// Grab a lock on the consensus set. Lock is demoted later in the function,
-	// failure to unlock before returning an error will cause a deadlock.
+func (cs *ConsensusSet) managedAcceptBlocks(blocks []types.Block) (bool, error) {
```

Comment:
I considered this interface, but it seems to be excessively complex (e.g., what to return in case of error in the middle of the group in the check for consecutiveness I added) but doesn't provide add information for the caller. If we encounter a non-fatal error (ErrBlockKnown, ErrNonExtendingBlock), we switch to next block; in case of a fatal error we don't consider subsequent blocks because they are children of a block causing a fatal error.

Basically we don't even need `bool` here (it is added in a separate commit but everything works before that commit). It was added to inform a caller (`managedReceiveBlocks`) if a blockchain was extended in case of a non-fatal error.